models/image.py

```diff
@@ -1,10 +1,133 @@
 
-class Image(Base):
     __tablename__ = 'images'
 
+    exposure_id = sa.Column(
+        sa.ForeignKey('exposures.id'),
+        nullable=True,
+        index=True,
+        doc=(
+            "ID of the exposure from which this image was derived. "
+            "Only set for single-image objects."
+        )
+    )
+
+    exposure = orm.relationship(
+        'Exposure',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Exposure from which this image was derived. "
+            "Only set for single-image objects."
+        )
+    )
+
+    source_images = orm.relationship(
+        'Image',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Images used to produce a multi-image object "
+            "(e.g., an images stack, reference, difference, super-flat, etc)."
+        )
+    )
+
+    is_multi_image = sa.Column(
+        sa.Boolean,
+        nullable=False,
+        default=False,
+        index=True,
+        doc=(
+            "Whether this image is a multi-image object. "
+            "Multi-image objects will have a null exposure, "
+            "and a list of Image objects from which it was derived. "
+        )
+    )
+
+    combine_method = sa.Column(
+        sa.Text,
+        nullable=True,
+        index=True,
+        doc=(
+            "Type of combination used to produce this multi-image object. "
+            "One of: coadd, subtraction. "
+        )
+    )
+
+    type = sa.Column(
+        sa.Text,
+        nullable=False,
+        index=True,
+        doc=(
+            "Type of image. One of: science, reference, difference, bias, dark, flat. "
+        )
+    )
+
+    provenance_id = sa.Column(
+        sa.ForeignKey('provenances.id'),
+        nullable=False,
+        index=True,
+        doc=(
+            "ID of the provenance of this image. "
+            "The provenance will contain a record of the code version"
+            "and the parameters used to produce this image. "
+        )
+    )
+
+    provenance = orm.relationship(
+        'Provenance',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Provenance of this image. "
+            "The provenance will contain a record of the code version"
+            "and the parameters used to produce this image. "
+        )
+    )
+
+    header = sa.Column(
+        JSONB,
+        nullable=False,
+        default={},
+        doc=(
+            "Header of the specific image for one section of the instrument. "
+            "Only keep a subset of the keywords, "
+            "and re-key them to be more consistent. "
+        )
+    )
+
+    mjd = sa.Column(
+        sa.Double,
+        nullable=False,
+        index=True,
+        doc=(
+            "Modified Julian date of the exposure (MJD=JD-2400000.5). "
+            "Multi-exposure images will have the MJD of the first exposure."
+        )
+    )
```

Comment:
We should probably define whether it's the MJD of the start of the exposure, or the MJD of the middle of the exposure.  MOST observatories give you the MJD of the start of the exposure in the header, but arguably the mean MJD is more useful.  Usually it doesn't matter because the timescale of the events we care about is enough longer than the typical exposure time that the difference is unimportant.  But, it's probably worth being clear about what we mean.

I'd probably be in favor of making it the mean MJD ; you could then also have the mean MJD for a coadded exposure.  (For a coadd, the MJD is a much more fraught concept, of course.)

Reply:
I'm going to add an `end_mjd` to each image, so that will track the start/end times of either a single exposure or a stack of images. The middle mjd will be the average of those two, by definition. And the exposure time is the sum of exposure times, regardless of whether you use sum or average, because that is just a scalar multiplication of the pixel values, it doesn't change the fact that the image contains photons from X seconds of exposure.